### PR TITLE
Adds TP properties and fixes barrack map purchasing.

### PR DESCRIPTION
**Pull request by celophi**
Note that purchased barrack maps are not persisted. This just fixes the charging.

### CODE REVIEW

File: system/db/barracks.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@
 //---------------------------------------------------------------------------
 
 [
-{ mapId: 11, name: "Cozy Lodge", price: 0, characters: 5 },
-{ mapId: 12, name: "Pyromancer Room", price: 5, characters: 7 },
-{ mapId: 13, name: "Banquet Hall", price: 5, characters: 10 },
+{ mapId: 11, name: "Cozy Lodge", price: 0, characters: 4 },
+{ mapId: 12, name: "Pyromancer Room", price: 69, characters: 4 },
+{ mapId: 13, name: "Banquet Hall", price: 69, characters: 4 },
```
Comment (celophi) [inline system/db/barracks.txt]: I seem to recall that each barrack had a larger available number of character slots, but the data file is not showing that.

File: src/Shared/Const/Message.cs [hunk 2]
```diff
@@ -71,4 +71,12 @@ public static class AddonMessage
 		/// </summary>
 		public const string KEYBOARD_TUTORIAL = "KEYBOARD_TUTORIAL";
 	}
+
+	/// <summary>
+	/// Constant strings sent with `BC_NORMAL_Run`.
+	/// </summary>
+	public static class BarrackMessage
+	{
+		public const string THEMA_BUY_SUCCESS = "THEMA_BUY_SUCCESS";
```
Comment (celophi) [inline src/Shared/Const/Message.cs]: @exectails: Once you have decided whether we should change the styling of these messages, we can change all of them at once.

File: src/LoginServer/World/Character.cs [hunk 3]
```diff
@@ -47,6 +47,7 @@ public Character()
 		{
 			this.Level = 1;
 			this.Channel = 1;
+			this.BarrackLayer = 1;
```
Comment (celophi) [inline src/LoginServer/World/Character.cs]: This is not directly related to the PR, but should be fixed anyway. It prevents a layer of '0' being saved to the database, and consequently, the character will not show up when `BC_COMMANDER_LIST` is sent.

Comment (exectails) [reply src/LoginServer/World/Character.cs]: If that's the case BarrackLayer should probably not even accept a 0 if that's not valid. The property could be clamped to 1~max, like I did for HP and co. in my last commit.

File: src/Shared/Const/Message.cs (hunk 2 above)
Comment (exectails) [reply src/Shared/Const/Message.cs]: Yea, I'm not sure. What do you think? Seeing upper case underscore naming makes me cringe, but it makes sense from the perspective of someone who digs around the client^^

Comment (celophi) [reply src/Shared/Const/Message.cs]: I think it's fine to rename them. Not everything needs to be the same as the client.
